.coveragerc

```diff
@@ -5,3 +5,7 @@ omit =
     */third_party/*
     */usr/*
     *_test.py
+    *core/tests/*
+    *main.py
```

Comment:
Why drop this and the next line?

Reply:
I thought these files are too specific to the app-engine and its working and that's why they cannot be tested like the other files

Reply:
I don't think such a principle applies. Which lines in those files are not currently covered?

Reply:
For example, these functions are not covered: [link](https://github.com/oppia/oppia/blob/c0218e13ba29f9bc25cc5ec6f7f13108ee4fdb9a/appengine_config.py#L32) and [link](https://github.com/oppia/oppia/blob/c0218e13ba29f9bc25cc5ec6f7f13108ee4fdb9a/main.py#L126)

Reply:
OK, thanks. Can you provide a full list of what is not covered (in the current versions of the files)? If we decide to exclude this I want to try and figure out a general principle for what should remain untested and why.

Reply:
Sure. You can find the exact coverage for [main.py](https://codeclimate.com/github/oppia/oppia/main.py/source) and [appengine_config.py](https://codeclimate.com/github/oppia/oppia/appengine_config.py/source)

Just enable the "coverage" check box. Thanks!

Reply:
Thanks! That's helpful.

For appengine_config.py, let's cover it but add a pragma to webapp_add_wsgi_middleware(), which I agree doesn't need a check. For line 86, swap out pil_path for something else in your tests; check sys.path at the end, and make sure to reset it at the end of the test. For line 108, swap out THIRD_PARTY_LIBS.

For main.py, we should cover it fully.

Reply:
Done